Assets/Project/Modules/Enemies/Turret/Scripts/scripts/ParabolicProjectile.cs

```diff
@@ -0,0 +1,193 @@
+        float yFinal = (float)(v0 * time * Math.Sin(angle) - (1f/2f) * -Physics.gravity.y * Mathf.Pow(time,2));
+        _line.SetPosition(count,_firePoint.position +direction * xFinal + Vector3.up * yFinal);
+    }
+
+    private void CalculatePath(Vector3 targetPos, float angle, out float v0, out float time)
+    {
+        float xt = targetPos.x;
+        float yt = targetPos.y;
+        float g = -Physics.gravity.y;
+
+        float v1 = Mathf.Pow(xt, 2) * g;
+        float v2 = 2 * xt * Mathf.Sin(angle) * Mathf.Cos(angle);
+        float v3 = 2 * yt * Mathf.Pow(Mathf.Cos(angle), 2);
+        v0 = Mathf.Sqrt(v1 / (v2 - v3));
+
+        time = xt / (v0 * Mathf.Cos(angle));
+
+    }
+    private async UniTaskVoid Movement(Vector3 direction, float v0,float angle,float time)
+    {
+        float t = 0;
+        while (t < time*2)
+        {
+            float x = v0 * t * Mathf.Cos(angle);
+            float y = (float)(v0 * t * Math.Sin(angle) - (1f/2f) * -Physics.gravity.y * Mathf.Pow(t,2));
+            transform.position = _firePoint.position + direction * x + Vector3.up * y;
+            
+            t += Time.deltaTime * speed;
+            await UniTask.NextFrame();
+        }
+        
+        
+    }
+
+    public float QuadraticEquation(float a, float b, float c, float sign)
+    {
+        return (-b + sign * Mathf.Sqrt(b * b - 4 * a * c)) / (2 * a);
+    }
+    private void CalculatePathWithHeight(Vector3 targetPos, float h, out float v0, out float angle, out float time)
+    {
+        float xt = targetPos.x;
+        float yt = targetPos.y;
+        float g = -Physics.gravity.y;
+
+        float b = Mathf.Sqrt(2 * g * h);
+        float a = (-0.5f * g);
+        float c = -yt;
+
+        float tPlus = QuadraticEquation(a, b, c, 1);
+        float tMinus = QuadraticEquation(a, b, c, -1);
+        time = tPlus > tMinus ? tPlus : tMinus;
+
+        angle = Mathf.Atan(b * time / xt);
+
+        v0 = b / Mathf.Sin(angle);
+
+
+    }
+
+    internal override void Init()
+    {
+        Invoke(nameof(Recycle),5);
+    }
+
+    private void OnCollisionEnter(Collision other)
+    {
+            _contactDamageHit.DamageSourcePosition = transform.position;
+            _contactDamageHit.UpdateKnockbackPushDirection(PositioningHelper.Instance.GetDirectionAlignedWithFloor(transform.position, other.transform.position));
+            _combatManager.TryDealDamage(other.gameObject, _contactDamageHit, out DamageHitResult damageHitResult);
+            _bulletBody.enabled = false;
+            RaycastHit hit;
+            if (Physics.Raycast(transform.position + Vector3.up, Vector3.down, out hit,_defaultProbingConfig.ProbeDistance,_defaultProbingConfig.CollisionLayerMask,_defaultProbingConfig.QueryTriggerInteraction))
+            {
+                var startRot = Quaternion.LookRotation(hit.normal) * Quaternion.Euler(new Vector3(0,90,90f));
+                Instantiate(_damageableArea, hit.point, startRot);
```

Comment:
In the next tasks, make both ParabolicProjectile and DamageableArea instancable via a Factory (ServiceLocator), which uses a Pool